File: solutions/HJK9810/2026-08-12/high/6782.py
import math

def algorithm():
    N = int(input())
    count = 0

    while N >= 2:
        if N == 2: break

        sqrt_N = math.sqrt(N)
        if sqrt_N == int(sqrt_N):
            N = sqrt_N
            count += 1
        else:
            next_sqrt = int(sqrt_N) + 1
            count += (next_sqrt ** 2 - N + 1)
            N = next_sqrt

    return count

T = int(input())
# 여러개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for test_case in range(1, T + 1):
    print(f"#{test_case} {algorithm()}")

File: solutions/HJK9810/2026-08-12/high/6782.java
import java.util.Scanner;
import java.io.FileInputStream;

class Solution
{
  public static void main (String args[]) throws Exception {
    Scanner sc = new Scanner(System.in);
    int T = sc.nextInt();

    for (int test_case = 1; test_case <= T; test_case++) {
      long N = sc.nextLong();
      long count = 0;

      while (N >= 2) {
        if (N == 2) break;

        double sqrtN = Math.sqrt(N);
        if ((long)sqrtN * (long)sqrtN == N) {
          N = (long) sqrtN;
          count += 1;
        } else {
          long nextPow = (long) sqrtN + 1;
          count += (nextPow * nextPow- N + 1);
          N = nextPow;
        }
      }

      System.out.printf("#%d %d\n", test_case, count);
    }
  }
}
